sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -276,13 +276,15 @@ _pi_event::_pi_event(pi_command_type type, pi_context context, pi_queue queue)
       isStarted_{false}, evEnd_{nullptr}, evStart_{nullptr}, evQueued_{nullptr},
       queue_{queue}, context_{context} {
 
-  if (is_native_event()) {
+  if (type != PI_COMMAND_TYPE_USER) {
     PI_CHECK_ERROR(cuEventCreate(&evEnd_, CU_EVENT_DEFAULT));
 
     if (queue_->properties_ & PI_QUEUE_PROFILING_ENABLE) {
       PI_CHECK_ERROR(cuEventCreate(&evQueued_, CU_EVENT_DEFAULT));
       PI_CHECK_ERROR(cuEventCreate(&evStart_, CU_EVENT_DEFAULT));
     }
+  } else {
+    cl::sycl::detail::pi::die("User-defined events not implemented");
```

Comment:
Would be good to give more granular message so that the user knows this error is coming from the CUDA backend.

Reply:
Fixed in  adcf8978b525aef45a5ffc888a42722017987c45